edb/edgeql/codegen.py

```diff
@@ -837,6 +837,24 @@ def _ddl_visit_bases(self, node: qlast.BasesMixin) -> None:
             self._write_keywords(' EXTENDING ')
             self.visit_list(node.bases, newlines=False)
 
+    PointerNode = TypeVar(
+        'PointerNode',
+        qlast.CreateConcretePointer,
+        qlast.CreateLink,
+        qlast.CreateProperty
+    )
+
+    def _ddl_add_bases_command(
+        self,
+        node: PointerNode,
+    ) -> PointerNode:
+        if node.bases:
+            return node.replace(commands=(  # type: ignore
```

Comment:
OK I think the actual fix for this is to update the definition of `replace` in `edb/common/ast/base.py` to have a generic type annotation on `self` and the return type.